libmattermost.c

```diff
@@ -970,6 +970,65 @@ mm_add_user_to_channel_response(MattermostAccount *ma, JsonNode *node, gpointer
 	mm_join_room(ma, channel);
 }
 
+// when Pidgin is open and a new channel is joined then add it to the blist
+static void
+mm_add_new_joined_channel_to_blist(MattermostAccount *ma, MattermostChannel *mm_channel)
+{
+	if (mm_channel_is_hidden(ma, mm_channel->id)) {
+		mm_g_free_mattermost_channel(mm_channel);
+	} else {
+		mm_set_group_chat(ma, mm_channel->team_id, mm_channel->name, mm_channel->id);
+		PurpleChat *chat = mm_purple_blist_find_chat(ma, mm_channel->id);
+
+		if (chat == NULL) {
+			GHashTable *defaults = g_hash_table_new_full(g_str_hash, g_str_equal, NULL, g_free);
+
+			const gchar *alias;
+			g_hash_table_insert(defaults, "team_id", g_strdup(mm_channel->team_id));
+			g_hash_table_insert(defaults, "id", g_strdup(mm_channel->id));
+			g_hash_table_insert(defaults, "creator_id", g_strdup(mm_channel->creator_id));
+			g_hash_table_insert(defaults, "type", g_strdup(mm_channel->type));
+			g_hash_table_insert(defaults, "display_name", g_strdup(mm_channel->display_name));

```

Comment:
Use constants.

Reply:
this is existing code where we have just made some tweaks to support our requirements, also such similar code is present at many places and Pidgin has not maintained constants for most of such things.

If we modify this code only we will also need to handle all such other cases for consistency which needs extra time, we can confirm from Mike if we can do refactoring of existing code.